process/block/shardblock.go

```diff
@@ -733,6 +733,18 @@ func (sp *shardProcessor) CommitBlock(
 		log.Debug(errNotCritical.Error())
 	}
 
+	for i := range finalHeaders {
+		rootHash := finalHeaders[i].GetRootHash()
+		if rootHash == nil {
+			continue
+		}
+
+		errNotCritical := sp.accounts.PruneTrie(rootHash)
```

Comment:
Can we add a test for the scenario where:
1. We have one block "BlockX" with a new root hash
2. A number of blocks are produced with no transactions (same root hash) 
3. "BlockX" becomes final so Prune is called

Expected results:
Root hash from block X is still in the trie, the Pruning does not remove it with its value

Reply:
Will add a todo, and will open another pull request with integration tests.